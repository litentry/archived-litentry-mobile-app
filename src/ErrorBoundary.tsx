import React from 'react';
import {Button, Layout, Text} from '@ui-kitten/components';
import globalStyles from 'src/styles';

export class ErrorBoundary extends React.Component<{}, {hasError: boolean}> {
  constructor(props: {}) {
    super(props);
    this.state = {hasError: false};
  }

  static getDerivedStateFromError() {
    // Update state so the next render will show the fallback UI.
    return {hasError: true};
  }

  componentDidCatch(error: any, errorInfo: any) {
    // You can also log the error to an error reporting service
    console.error('ERROR CAUGHT: ');
    console.error(error);
    console.error(errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return (
        <Layout style={globalStyles.centeredContainer}>
          <Text>Something went wrong!</Text>
          <Button onPress={() => this.setState({hasError: false})}>Try again</Button>
        </Layout>
      );
    }

    return this.props.children;
  }
}
