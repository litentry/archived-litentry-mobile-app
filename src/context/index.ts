export * from './AccountsContext';
export * from './ChainApiContext';
export * from './InAppNotificationContext';
export * from './NetworkContext';
export * from './NavigationContainer';
export * from './ThemeContext';
