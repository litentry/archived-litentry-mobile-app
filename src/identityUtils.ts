import {IdentityJudgement} from '@polkadot/types/interfaces';

export const mapStatusText = (judgement: IdentityJudgement, hasParent: boolean) => {
  if (judgement.isErroneous) {
    return {text: 'Erroneous', category: 'bad', icon: 'alert-circle-outline'};
  }

  // if (judgement.isFeePaid) {
  //   return {text: 'Fee Paid', category: 'pending'};
  // }
  //
  // if (judgement.asFeePaid) {
  //   return {text: 'Pending', category: 'pending'};
  // }

  if (judgement.isReasonable) {
    return {text: 'Reasonable', category: 'good', icon: hasParent ? 'link' : 'check-circle'};
  }

  if (judgement.isKnownGood) {
    return {text: 'Known Good', category: 'good', icon: hasParent ? 'link' : 'check-circle'};
  }

  if (judgement.isOutOfDate) {
    return {text: 'Out of Date', category: 'bad', icon: 'alert-circle'};
  }

  if (judgement.isLowQuality) {
    return {text: 'Low Quality', category: 'bad', icon: 'alert-circle'};
  }

  return {text: 'Unknown', category: 'neutral', icon: 'help-circle'};
};
