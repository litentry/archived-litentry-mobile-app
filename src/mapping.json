{
  "strict": {
    "text-font-family": "DMSans-Regular",

    "text-heading-3-font-size": 45,
    "text-heading-3-font-weight": "400",

    "text-heading-4-font-size": 33,
    "text-heading-4-font-weight": "400",

    "text-heading-5-font-size": 24,
    "text-heading-5-font-weight": "400",

    "text-heading-6-font-size": 20,
    "text-heading-6-font-weight": "400",

    "text-label-font-size": 10,
    "text-label-font-weight": "400",

    "text-subtitle-1-font-size": 15,
    "text-subtitle-2-font-size": 13,

    "text-caption-1-font-size": 12
  }
}
