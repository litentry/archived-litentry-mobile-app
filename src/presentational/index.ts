export {Chart} from './Chart';
export {noTopEdges, default as SafeView} from './SafeView';
