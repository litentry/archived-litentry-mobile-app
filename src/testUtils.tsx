/* eslint-disable no-restricted-imports */
import React from 'react';
import {render, RenderOptions} from '@testing-library/react-native';
import {ApolloProvider, ApolloClient, InMemoryCache, HttpLink} from '@apollo/client';
import fetch from 'cross-fetch';

import ThemeProvider from 'context/ThemeContext';

export const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: new HttpLink({uri: 'http://localhost:3000/graphql', fetch}),
});

function Providers({children}: {children: React.ReactNode}) {
  return (
    <ThemeProvider>
      <ApolloProvider client={client}>{children}</ApolloProvider>
    </ThemeProvider>
  );
}

type CustomRenderOptions = Omit<RenderOptions, 'wrapper'>;

function customRender(ui: React.ReactElement, options?: CustomRenderOptions) {
  return render(ui, {wrapper: Providers, ...options});
}

const mockedNavigation = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: mockedNavigation,
    }),
  };
});

export * from '@testing-library/react-native';
export {customRender as render};
