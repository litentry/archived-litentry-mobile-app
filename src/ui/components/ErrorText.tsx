import React from 'react';
import {StyleSheet} from 'react-native';
import {Text, useTheme} from '@ui/library';

export function ErrorText(props: {children: React.ReactNode}) {
  const theme = useTheme();
  return <Text style={[styles.errorText, {color: theme.colors.error}]}>{props.children}</Text>;
}

const styles = StyleSheet.create({
  errorText: {
    textAlign: 'center',
  },
});
