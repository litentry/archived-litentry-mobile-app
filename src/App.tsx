import React from 'react';
import {Host} from 'react-native-portalize';
import {IconRegistry} from '@ui-kitten/components';
import {EvaIconsPack} from '@ui-kitten/eva-icons';
import ThemeProvider from './context/ThemeContext';
import ChainApiContextProvider from 'context/ChainApiContext';
import {AccountsProvider} from 'src/context/AccountsContext';
import TxContextProvider from 'context/TxContext';
import AppNavigator from 'src/navigation/AppNavigator';
import {IonicIconsPack} from './Ionic-icons';
import {ErrorBoundary} from 'src/ErrorBoundary';
import {QueryClient, QueryClientProvider} from 'react-query';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import NetworkContextProvider from 'src/context/NetworkContext';
import InAppNotificationContextProvider from 'src/context/InAppNotificationContext';

// init type registry
import 'src/typeRegistry';

const queryClient = new QueryClient();

export default function App() {
  return (
    <>
      <IconRegistry icons={[EvaIconsPack, IonicIconsPack]} />
      <NetworkContextProvider>
        <ChainApiContextProvider>
          <QueryClientProvider client={queryClient}>
            <ThemeProvider>
              <InAppNotificationContextProvider>
                <SafeAreaProvider>
                  <ErrorBoundary>
                    <Host>
                      <AccountsProvider>
                        <TxContextProvider>
                          <AppNavigator />
                        </TxContextProvider>
                      </AccountsProvider>
                    </Host>
                  </ErrorBoundary>
                </SafeAreaProvider>
              </InAppNotificationContextProvider>
            </ThemeProvider>
          </QueryClientProvider>
        </ChainApiContextProvider>
      </NetworkContextProvider>
    </>
  );
}
