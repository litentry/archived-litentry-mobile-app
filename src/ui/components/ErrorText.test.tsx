/* eslint-disable @typescript-eslint/no-var-requires */
import React from 'react';
import {create} from 'react-test-renderer';
import {render} from 'src/testUtils';
import {ErrorText} from './ErrorText';

jest.doMock('./ErrorText', () => {
  return {
    ErrorTextMockComponent: jest.fn(() => <ErrorText>Mocked Error Text</ErrorText>),
  };
});
const {ErrorTextMockComponent} = require('./ErrorText');

describe('ErrorText component', () => {
  it('should render the prop component correctly', async () => {
    const {getByTestId} = render(<ErrorTextMockComponent />);
    expect(getByTestId('error_text').children.at(0)).toEqual('Mocked Error Text');
  });

  it('Testing render with snapshot', () => {
    const errorTextComponent = create(<ErrorTextMockComponent />);
    const tree = errorTextComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
