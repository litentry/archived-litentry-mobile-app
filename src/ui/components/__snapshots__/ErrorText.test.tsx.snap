// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorText component Testing render with snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      Object {
        "textAlign": "left",
      },
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#B00020",
        },
      ],
    ]
  }
  testID="error_text"
>
  Mocked Error Text
</Text>
`;
